src/components/ListItem.jsx

```diff
@@ -1,9 +1,46 @@
 import './ListItem.css';
+import { useEffect } from 'react';
+import { updateItem } from '../api/firebase';
+
+export function ListItem({ listToken, item }) {
+	const millisecondsIn24hrs = 84600000;
+	useEffect(() => {
+		if (item.isChecked) {
+			let calcDate =
+				item.dateLastPurchased.seconds * 1000 + millisecondsIn24hrs;
+
+			setTimeout(() => {
+				updateItem(listToken, item, {
+					isChecked: !item.isChecked,
+				});
+			}, calcDate);
+		}
+	}, [item, listToken]);
+	//
+	const handleChange = () => {
```

Comment:
Functions that depend on variables like state and props should be mutated by using the `useCallback` of react